src/model/chat.ts

```diff
@@ -131,12 +131,13 @@ export class ChatModel extends AbstractModel<
       const choice = chunk.choices[0];
       const response: Model.Chat.ApiResponse = {
         ...chunk,
+        object: 'chat.completion',
         choices: [
           {
             finish_reason:
               choice.finish_reason as Model.Chat.Response['choices'][0]['finish_reason'],
             index: choice.index,
-            message: choice.delta as Model.Message,
+            message: choice.delta as Model.Message & { role: 'assistant' },
```

Comment:
Do we need this now that you fixed the types for `openai-fetch`?

Reply:
Good question && yes, unfortunately TS complains without it.

One thing I've noticed is that the new `openai` types seem to have gotten a lot stricter in terms of inputs/outputs used in different places. I could be wrong since I've only just started getting back into things.

Reply:
<img width="490" alt="CleanShot 2023-11-08 at 14 45 45@2x" src="https://github.com/dexaai/dexter/assets/552829/27611493-3266-4a67-a7fd-45ccbcb6f298">

Reply:
i think this is just from their jank types so we can leave as is for now